### PR TITLE
Correctly skip already-deleted edges when enqueuing deletions

### LINKED ISSUE #2230: Wrong number of deleted relationships is reported
RedisGraph 2.8.9

First, let's create a graph with 2 relationships:

```
graph.DELETE test
graph.QUERY test "CREATE (x1:A)-[r:R]->(n2:B)-[t:T]->(x1)"
```

"Relationships created: 2"

Now let's delete these relationships as follows:

```
graph.QUERY test "MATCH ()-[r]-() delete r delete r,r delete r,r"
```

"Relationships deleted: 6"

Though only 2 relationships were deleted.

### PR DESCRIPTION
**Pull request by jeffreylovitz**
Resolves #2230

### CODE REVIEW

File: src/graph/graph.c [hunk 1]
```diff
@@ -1104,6 +1104,8 @@ void Graph_BulkDelete(Graph *g, Node *nodes, uint node_count, Edge *edges, uint
 			// As long as current is the same as follows.
 			while(i < edge_count - 1 && ENTITY_GET_ID(edges + i) == ENTITY_GET_ID(edges + i + 1)) i++;
 
+			// skip edges that have already been deleted
+			if(!DataBlock_GetItem(g->edges, ENTITY_GET_ID(edges + i))) continue;
 			if(uniqueIdx < i) edges[uniqueIdx] = edges[i];
```
Comment (swilly22) [inline src/graph/graph.c:1109]: is it possible for `uniqueIdx` to be greater than `i` ?
if not, let's remove this `if` statement.

File: src/graph/graph.c [hunk 2]
```diff
@@ -1104,6 +1104,8 @@ void Graph_BulkDelete(Graph *g, Node *nodes, uint node_count, Edge *edges, uint
 			// As long as current is the same as follows.
 			while(i < edge_count - 1 && ENTITY_GET_ID(edges + i) == ENTITY_GET_ID(edges + i + 1)) i++;
 
+			// skip edges that have already been deleted
+			if(!DataBlock_GetItem(g->edges, ENTITY_GET_ID(edges + i))) continue;
```
Comment (swilly22) [inline src/graph/graph.c:1108]: Consider switching to `DataBlock_ItemIsDeleted`

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: src/graph/graph.c (hunk 2 above)
Comment (jeffreylovitz) [reply src/graph/graph.c:1108]: These edges aren't pointers into the DataBlock; they are `SIValue` pointer types retrieved by calling `AR_EXP_Evaluate`.

Comment (swilly22) [reply src/graph/graph.c:1108]: is `(edges + i)->entity` set?
if so we should be able to call `DataBlock_ItemIsDeleted((Entity*)(edges + i));`

Comment (jeffreylovitz) [reply src/graph/graph.c:1108]: `entity` is set, but I don't think that has any relation to `DataBlock_ItemIsDeleted`. That function expects the bytes before the pointer to be a `DataBlockItemHeader`, which is not true in this case.

**Review by AviAvni — APPROVED** [review]

**Pull request merged**